### PR TITLE
Add crypto operations for browser-storage key

### PR DESCRIPTION
**Pull request by nmattia**
This introduces the cryptographic operations for storing an II keypair in browser storage.

The keypair is encrypted twice (once with a generated key, once with a key derived from a user PIN, see module) before being stored in IndexedDB.

The test-setup's crypto module is updated. The previous crypto module did not support `CryptoKey`.

### CODE REVIEW

File: src/frontend/src/crypto/pinIdentity.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+}: {
+  pin: string;
+}): Promise<{
+  identity: SignIdentity;
+  pinIdentityMaterial: PinIdentityMaterial;
+}> => {
+  const browserIv = window.crypto.getRandomValues(new Uint8Array(96));
+  const pinIv = window.crypto.getRandomValues(new Uint8Array(96));
+  const pinSalt = window.crypto.getRandomValues(new Uint8Array(96));
+  const pinPbkdfIters: number = 10000;
```
Comment (nmattia) [inline src/frontend/src/crypto/pinIdentity.ts:54]: Please double check these values

File: src/frontend/src/crypto/pinIdentity.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  const keypairNamedCurve: NistEc = "P-256";
+
+  const keypair = await generateKeyPair({ namedCurve: keypairNamedCurve });
+  const secretKey = keypair.getKeyPair().privateKey;
+
+  const browserKey = await generateBrowserKey();
+  const encryptedOnce = await crypto.subtle.wrapKey(
+    "pkcs8" /* the export format */,
```
Comment (nmattia) [inline src/frontend/src/crypto/pinIdentity.ts:62]: Please double check this, not sure if this or jwk is better

File: src/frontend/src/crypto/pinIdentity.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
+import { SignIdentity } from "@dfinity/agent";
+import { ECDSAKeyIdentity } from "@dfinity/identity";
+import { z } from "zod";
```
Comment (frederikrothenberger) [inline src/frontend/src/crypto/pinIdentity.ts:3]: If we start using the library in our code, shouldn't we have a direct dependency on it too?

File: src/frontend/src/crypto/pinIdentity.ts (hunk 2 above)
Comment (frederikrothenberger) [reply src/frontend/src/crypto/pinIdentity.ts:62]: I don't think this matters here, as long as we are consistent when importing and exporting.

**Review by frederikrothenberger — APPROVED** [review]
This looks good to me, thanks! Left a few minor comments. 

I would recommend trying to bump the number of PBKDF2 iterations by ~10x (maybe more, but requires testing on older phones to see how much delay it causes) to make brute-forcing more expensive.

This definitely requires a second approval by prod-sec and / or the crypto team.

File: src/frontend/src/crypto/pinIdentity.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/crypto/pinIdentity.ts]: bumped to 10000, we'll see what happens when we do some manual testing before releasing

**Comment by nmattia** [comment]
@tmu0 addressed here: https://github.com/dfinity/internet-identity/pull/1861

**Pull request merged**